src/Relation.js

```diff
@@ -193,6 +193,9 @@ utils.addHiddenPropsToTarget(Relation.prototype, {
 
   // e.g. user hasMany post via "foreignKey", so find all posts of user
   findExistingLinksByForeignKey (id) {
+    if (id === undefined || id === null) {
+      return
+    }
```

Comment:
or we could rework this method such that it returns an empty array in this case.